source/MaterialXRuntime/RtPath.cpp

```diff
@@ -0,0 +1,70 @@
+//
+// TM & (c) 2019 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXRuntime/RtPath.h>
+
+#include <MaterialXRuntime/Private/PrvPath.h>
+
+namespace MaterialX
+{
+
+RtPath::RtPath(RtObject obj) :
+    _ptr(nullptr)
+{
+    if (obj.hasApi(RtApiType::ELEMENT))
+    {
+        PrvElement* elem = obj.data()->asA<PrvElement>();
+        PrvElement* root = elem->getRoot();
+        _ptr = new PrvPath(root->shared_from_this(), elem->shared_from_this());
+    }
+}
+
+RtPath::~RtPath()
+{
+    delete static_cast<PrvPath*>(_ptr);
+}
+
+bool RtPath::isValid() const
+{
+    return static_cast<PrvPath*>(_ptr)->isValid();
```

Comment:
Will this crash if a user doesn't have an element but some other type? Obviously that's a bug, I just dunno if crashing is best or not.

Reply:
Similar comment for some of the functions below.

Reply:
Good catch! In practice it will currently never happen since all objects in a stage are derived from PrvElement. But if that changes in the future we might run into a crash here. I'll try to make this more safe.